exporter/api.py

```diff
@@ -0,0 +1,111 @@
+"""
+Functions for exporting learning resources
+"""
+
+from __future__ import unicode_literals
+
+from tempfile import mkdtemp, mkstemp
+import tarfile
+import os
+from shutil import rmtree
+import re
+
+from django.utils.text import slugify
+from django.core.files.storage import default_storage
+from lore.settings import EXPORT_PATH_PREFIX
+
+
+def export_resources_to_directory(learning_resources):
+    """
+    Create files of LearningResource and StaticAsset contents inside directory.
+
+    Args:
+        learning_resources (list of learningresources.models.LearningResource):
+            LearningResources to export in tarball
+    Returns:
+        unicode: Newly created temp directory with files inside of it.
+    """
+    def sanitize(title):
+        """Sanitize title for use in filename."""
+        # Limit filename to 200 characters since limit is 256
+        # and we have a number and extension too.
+        return slugify(title)[:200]
+
+    tempdir = mkdtemp()
+    try:
+        for learning_resource in learning_resources:
+            filename = "{id}_{title}.xml".format(
+                id=learning_resource.id,
+                title=sanitize(learning_resource.title),
+            )
+            type_name = learning_resource.learning_resource_type.name
+            try:
+                os.mkdir(os.path.join(tempdir, type_name))
+            except OSError:
+                pass
+            with open(os.path.join(tempdir, type_name, filename), 'w') as f:
+                f.write(learning_resource.content_xml)
+
+            for static_asset in learning_resource.static_assets.all():
+                # Output to static directory.
+                static_path = re.sub(r"^assets/", "static/",
+                                     static_asset.asset.name)
+
+                relative_path, filename = os.path.split(static_path)
+                try:
+                    os.makedirs(os.path.join(tempdir, relative_path))
+                except OSError:
```

Comment:
Seems reasonable. It's unfortunate that `FileExistsError` only exists in Python 3